server/src/main/java/org/elasticsearch/search/aggregations/support/ValuesSourceConfig.java

```diff
@@ -101,7 +102,11 @@
             } else if (indexFieldData instanceof IndexGeoPointFieldData) {
                 config = new ValuesSourceConfig<>(ValuesSourceType.GEOPOINT);
             } else {
-                config = new ValuesSourceConfig<>(ValuesSourceType.BYTES);
+                if (fieldType instanceof RangeFieldMapper.RangeFieldType) {
```

Comment:
Maybe promote this up to an `else if`?  Not sure if it'd be cleaner or not, but it's nice to reduce nested if where possible.

Reply:
Sure.  The reason I had an extra level of nesting here was to emphasize that we were checking `fieldType` instead of `indexFieldData` like the other clauses do, but reading it now I'm not sure that's necessary to call out at all, and even if it is, a comment would do better than some fancy-pants nesting.